experimental/coinbase-negotiator/README.md

```diff
@@ -0,0 +1,168 @@
+one/ones)
+
+#### Pros:
+1. pool is noncustodial
+2. miners can verify the 2 below prop of the coinbase tx that they are mining (a1 is the output of the coinbase tx)
+    1. they can see that a valid transaction from a1 to the miner address exist and have the *right*
+       amount
+    2. they need to sign each valid tx from a1 so they can see that the total amount of all the
+       presigned txs from a1 is (coinbase input - the fair reward that they expect)
+
+#### Cons:
+1. not scale  well on chain, cause if you are a small miner and you get paid at each found
+   blocks you will likely end up paying a lot in fees. You can mitigate it (1) sending the txs when the
+   fee is small, (2) aggregating txs, (3) the pool is not paying you at each founded block but only
+   when you will be paid an amount above a minimum one.
+2. not scale well off chain: pool need to coordinate all the miners, and everyone need to agree and be
+   online if only one fail the process need to be restarted from zero.
+
+### CTV
+Using CTV you will have the above pros without the above cons so it make sense to add this
+possibility if it will be merged on master.
+
+### No coinbase negotiation
+Pool can reserve a path to itself where they do custody bitcoin for miner that prefer to use a custodian pool.
+
+### Messages
+
+#### Valid for both Schnorr and CTV negotiations:
+
+Upstream will use the provided address as receiving address for the downstream path.
+```
+Client -> Server
+NewAddress:
+    address: B0255
+    
+msg_type: 0x77
+```
+
+#### Valid for Schnorr negotiations:
+
+
+Upstream will use the pub keys as Xi and Ri to calculate X and R as described here
+https://blog.blockstream.com/en-musig-key-aggregation-schnorr-signatures/ (MuSig paragrapher)
+```
+Client -> Server
+NewPubKeyPair:
+    pub_key_x: B0255
+    pub_key_r: B0255
+
+msg_type: 0x78
+```
+
+
+
+Downstream will use X, R, m, and L to sign m as described here
+https://blog.blockstream.com/en-musig-key-aggregation-schnorr-signatures/ (MuSig paragrapher)
+```
+Server -> Client
+NewTxToSign:
+    x: B0255
+    r: B0255
+    l: B0255
+    m: B064K
+
+
+msg_type: 0x79
+```
+
+
+Upstream will use the provided si value to calculate s as described here
+https://blog.blockstream.com/en-musig-key-aggregation-schnorr-signatures/ (MuSig paragrapher)
+```
+Client -> Server
+NewSignature:
+    s: B0255
+```
+
+
+After that a tx has been signed by all the downstream, upstream compute the valid signature s and
+send it to the downstream that control the output in m
+```
+Server -> Client
+ValidSignature:
+   s: B0255
+
+
+msg_type: 0x7A
+```
+
+
+When downstream will start to mine a new block it will use the new coinbase tx if it know a valid tx
+that spend bitcoin from the coinbase output to the downstream's controlled address, and if the sum
+of all the bitcoin spent by the tx signed via NewTxToSign do not exceed the coinbase input
```

Comment:
When the downstream nodes (clients) start to mine a new block, they will each use the new coinbase transaction if:
1. the coinbase output yields bitcoin spendable by the miner provided address, or 
2. if the sum of the bitcoin spent by the transaction signed via `NewTxToSign` does not exceed the coinbase input.

Reply:
Should briefly explain the significance of signing via `NewTxToSign`.